### PR TITLE
With minikube

### LINKED ISSUE #12: Demo does not work ATM
The demo does currently not work until the following two issues are resolved:

- [ ] https://github.com/kubevirt/kubevirt/issues/316 -- We especially need the part where we start to use pod networking
- [x] https://github.com/kubernetes/minikube/issues/1845 -- We also need macvtap support in minikube for networking

### PR DESCRIPTION
**Pull request by fabiand**
This PR moves the demo away from a VM based setup to minikube.

So far we were not able to deploy KubeVirt on minikube, but with recent changes this is now possible.

Resolves #12 #11 #10 #9 #2

### CODE REVIEW

**Comment by fabiand** [comment]
Networking will still be rough, this can be tested by making sure that the VM can be launched and is running.

**Comment by itamarh** [comment]
Gave this a try on latest Fedora 26, but it still has kubernetes 1.6.7.
easy enough to update to 1.7.x via:
sudo dnf update kubernetes-client --enablerepo=updates-testing
but i wonder if we are going to be version sensitive... and geared for minikube / isolating the host, could we run the kubectl from a container...?

**Comment by fabiand** [comment]
@itamarh thanks for testing - did it actually work for you?

In general we aim to be not version sensitive, however, in right now we are a little sensitive because the Kubernetes extension mechanism has changed (from TPR to CRD/UAS) and this requires some client support. Once that has passed, we should be more stable wrt client side requirements.

The idea to run `kubectl` from a container is nice, and might be easy to achieve - Opened #15 to track this.

**Comment by itamarh** [comment]
yes, it actually worked for me.
"Once that has passed, we should be more stable wrt client side requirements." - famous last words?
Its just such a fast moving technology...

**Comment by fabiand** [comment]
:)

Great, thanks for testing @itamarh

File: README.md [hunk 1]
```diff
@@ -1,144 +1,110 @@
...
+>   --iso-url https://storage.googleapis.com/minikube-builds/1846/minikube-testing.iso
+> ```
 
-You need to install the follwing tools:
 
-- `make`
-- `qemu-kvm`
-- `virt-builder`
-- `expect` (for `make check`)
+1. If not installed, install minikube as described here:
```
Comment (rmohr) [inline README.md]: The requirement for kubectl could be mentioned too (instructions are in the minikube setup guide too where to get it, but it is not so easy to see).

File: README.md [hunk 2]
```diff
@@ -1,144 +1,110 @@
...
+- Checking kubectl version ... OK
+- Fetching and registering virtctl ... OK
 
-## Check (optional)
+# Now the plugin is ready to use
 
-If you want, then you can use `make check` to run a minimal integration test.
-This will create a VM and check if it's really getting created in libvirtd.
+# Connect to the serial console
+$ kubectl plugin virt -- console -s http://$(minikube ip):8184 testvm
```
Comment (rmohr) [inline README.md]: The -s is not so nice. We could proxy the whole traffic throught 8184. Maybe we should create a second config file based on the provided one and set this server address. Undeploy could undo that change too ...

Comment (fabiand) [reply README.md]: Stu's work should solve this, thus I'd not do a workaround here atm.

File: README.md (hunk 1 above)
Comment (fabiand) [reply README.md]: Yes - I'll add a few words (tho not the complete setup guide).

Comment (rmohr) [reply README.md]: Perfect, exactly what I had in mind.

Comment (fabiand) [reply README.md]: done.

Comment (rmohr) [reply README.md]: The driver is a good hint too, what I meant is that `kubectl` needs to be installed separately.

**Comment by fabiand** [comment]
Cool - Works for two, thus merging.

**Pull request merged**